src/packet_analysis/protocol/tcp/TCPSessionAdapter.cc

```diff
@@ -407,7 +417,14 @@ static zeek::RecordValPtr build_syn_packet_val(bool is_orig, const zeek::IP_Hdr*
 	v->Assign(5, winscale);
 	v->Assign(6, MSS);
 	v->Assign(7, static_cast<bool>(SACK));
-
+	if(TSval < 0)
+		v->Assign(8, (int) TSval);
+	else
+		v->Assign(8, (uint32_t) TSval);
```

Comment:
It appears that `int32_t` isn't supported by `Assign` so I had to use this. I'm assuming this could be added to `Val.h`? Feedback welcomed :P

Reply:
Couple of things here:

- We shouldn't mix the types being assigned to field 8 (`int`/`uint32`) because that would lead to inconsistent typing in script-land.
- If we turn the script-side fields into optional counts as suggested above, we can cast to `uint64_t` here and then set only  in the case of being >= 0.